src/Export/Field/ImageUrl.php

```diff
@@ -18,16 +18,29 @@ public function getName(): string
 
     public function getValue(Entity $entity): string
     {
-        $media = $entity->getMedia();
-        return $media
-            ? method_exists($media, 'first')
-                ? $media->first()->getMedia()->getUrl()
-                : $media->getUrl()
-            : '';
+        return $this->checkAndReturnMediaUrlOrEmptyString($entity);
     }
 
     public function getCompatibleEntityTypes(): array
     {
         return [SalesChannelProductEntity::class, CategoryEntity::class, ProductManufacturerEntity::class];
     }
+
+    private function checkAndReturnMediaUrlOrEmptyString(?Entity $entity = null): string
+    {
+        $media = $entity->getMedia();
+
+        if (!$media) {
```

Comment:
can we rework this a little bit to have at lease one `return ''`  3 returns in one method doesn't sound well, however i like very much the method name

Reply:
I've trying but to be honest I can't see too much possibilities to rewrite it, we have 4 possible cases:
1) media !instanceof Entity - return empty string
2) !method_exists($media, 'first') -  return $media->getUrl() 
3) method_exists($media, 'first') && $media->first() - return $media->first()->getMedia()->getUrl()
4) method_exists($media, 'first') && !$media->first() - return empty string

So we need to check 4 conditions and serve each of them, and return appropriate result according to condition.

Maybe You have any idea? Because 4 returns could looks a bit terrifying but anyway it is according to PSR

Reply:
I think we can use a power of recursion here. There is some pattern in our actions. 
*  Check if entity has `getMedia`
* * yes - check if returned value from `getMedia` has `first`
* * * yes run process again with returned value of `first`
* * * no - we have our MediaEntity from which we can take the URL
* * no - throw exception

Reply:
I think we can use a power of recursion here. There is some pattern in our actions.

* Check if entity has getMedia
* * yes - check if returned value from getMedia has first
* * * yes run process again with returned value of first
* * * no - we have our MediaEntity from which we can take the URL
* * no - throw exception

Reply:
I have add a recursion to the checking method but I'm not added exception throwing bcs missing image is not an exception, you can remove it in the admin